napi_js_props.md

```diff
@@ -0,0 +1,225 @@
+# N-API - JavaScript Data Types and Values 
+
+N-API exposes a set of APIs to interrogate and set properties on JavaScript objects.
+Some of these types are documented under [Section 7](https://tc39.github.io/ecma262/#sec-operations-on-objects) 
+of the [ECMAScript Language Specification](https://tc39.github.io/ecma262/).
+
+Fundamentally, all properties in N-API can be represented in one of the following forms:
+- Named: a simple utf8 encoded string
+- Integer-Indexed: an index value represented by `uint32_t`
+- JavaScript Object: a complex JavaScript Object represented by `napi_value`. This is the case if the key for a particular property is not a String or Number.
```

Comment:
... and it should specifically list numbers, strings, and symbols, and nothing else, because AFAIK you can't have Object- or Array-valued keys - they get coerced to strings, and besides, that's what (Weak)?Map objects are for.